diff --git a/board/freescale/mx6ullevk/imximage.cfg b/board/freescale/mx6ullevk/imximage.cfg
index 4604b62..bcea07f 100644
--- a/board/freescale/mx6ullevk/imximage.cfg
+++ b/board/freescale/mx6ullevk/imximage.cfg
@@ -61,29 +61,31 @@ DATA 4 0x020c4080 0xffffffff
 
 DATA 4 0x020E04B4 0x000C0000
 DATA 4 0x020E04AC 0x00000000
-DATA 4 0x020E027C 0x00000030
-DATA 4 0x020E0250 0x00000030
-DATA 4 0x020E024C 0x00000030
-DATA 4 0x020E0490 0x00000030
-DATA 4 0x020E0288 0x000C0030
+DATA 4 0x020E027C 0x00000028
+DATA 4 0x020E0250 0x00000028
+DATA 4 0x020E024C 0x00000028
+DATA 4 0x020E0490 0x00000028
+DATA 4 0x020E0288 0x000C0028
 DATA 4 0x020E0270 0x00000000
-DATA 4 0x020E0260 0x00000030
-DATA 4 0x020E0264 0x00000030
-DATA 4 0x020E04A0 0x00000030
+DATA 4 0x020E0260 0x00000028
+DATA 4 0x020E0264 0x00000028
+DATA 4 0x020E04A0 0x00000028
 DATA 4 0x020E0494 0x00020000
-DATA 4 0x020E0280 0x00000030
-DATA 4 0x020E0284 0x00000030
+DATA 4 0x020E0280 0x00000028
+DATA 4 0x020E0284 0x00000028
 DATA 4 0x020E04B0 0x00020000
-DATA 4 0x020E0498 0x00000030
-DATA 4 0x020E04A4 0x00000030
-DATA 4 0x020E0244 0x00000030
-DATA 4 0x020E0248 0x00000030
+DATA 4 0x020E0498 0x00000028
+DATA 4 0x020E04A4 0x00000028
+DATA 4 0x020E0244 0x00000028
+DATA 4 0x020E0248 0x00000028
 DATA 4 0x021B001C 0x00008000
 DATA 4 0x021B0800 0xA1390003
-DATA 4 0x021B080C 0x00000004
-DATA 4 0x021B083C 0x41640158
-DATA 4 0x021B0848 0x40403237
-DATA 4 0x021B0850 0x40403C33
+DATA 4 0x021B080C 0x00000000
+DATA 4 0x021B0810 0x001F001F
+DATA 4 0x021B083C 0x015C0158
+DATA 4 0x021B0840 0x00000000
+DATA 4 0x021B0848 0x40402E32
+DATA 4 0x021B0850 0x40403C36
 DATA 4 0x021B081C 0x33333333
 DATA 4 0x021B0820 0x33333333
 DATA 4 0x021B082C 0xf3333333
@@ -93,9 +95,9 @@ DATA 4 0x021B08b8 0x00000800
 DATA 4 0x021B0004 0x0002002D
 DATA 4 0x021B0008 0x1B333030
 DATA 4 0x021B000C 0x676B52F3
-DATA 4 0x021B0010 0xB66D0B63
+DATA 4 0x021B0010 0xB66E0E63
 DATA 4 0x021B0014 0x01FF00DB
-DATA 4 0x021B0018 0x00201740
+DATA 4 0x021B0018 0x00211740
 DATA 4 0x021B001C 0x00008000
 DATA 4 0x021B002C 0x000026D2
 DATA 4 0x021B0030 0x006B1023
@@ -104,11 +106,11 @@ DATA 4 0x021B0000 0x84180000
 DATA 4 0x021B0890 0x00400000
 DATA 4 0x021B001C 0x02008032
 DATA 4 0x021B001C 0x00008033
-DATA 4 0x021B001C 0x00048031
+DATA 4 0x021B001C 0x00448031
 DATA 4 0x021B001C 0x15208030
 DATA 4 0x021B001C 0x04008040
-DATA 4 0x021B0020 0x00000800
-DATA 4 0x021B0818 0x00000227
+DATA 4 0x021B0020 0x00007800
+DATA 4 0x021B0818 0x00000337
 DATA 4 0x021B0004 0x0002552D
 DATA 4 0x021B0404 0x00011006
 DATA 4 0x021B001C 0x00000000
diff --git a/board/freescale/mx6ullevk/mx6ullevk.c b/board/freescale/mx6ullevk/mx6ullevk.c
index fcb1090..683687f 100644
--- a/board/freescale/mx6ullevk/mx6ullevk.c
+++ b/board/freescale/mx6ullevk/mx6ullevk.c
@@ -37,6 +37,7 @@
 #endif
 #endif /*CONFIG_FSL_FASTBOOT*/
 
+#define CONFIG_MX6ULL_EVK_EMMC_REWORK
 DECLARE_GLOBAL_DATA_PTR;
 
 #define UART_PAD_CTRL  (PAD_CTL_PKE | PAD_CTL_PUE |		\
@@ -184,7 +185,6 @@ static void iox74lv_init(void)
 void iox74lv_set(int index)
 {
 	int i;
-
 	for (i = 7; i >= 0; i--) {
 		gpio_direction_output(IOX_SHCP, 0);
 
@@ -241,7 +241,8 @@ static struct i2c_pads_info i2c_pad_info1 = {
 #define I2C_PMIC       0
 int power_init_board(void)
 {
-	if (is_mx6ull_9x9_evk()) {
+	//if (is_mx6ull_9x9_evk()) {
+	if (1) {
 		struct pmic *pfuze;
 		int ret;
 		unsigned int reg, rev_id;
@@ -258,8 +259,7 @@ int power_init_board(void)
 		pmic_reg_read(pfuze, PFUZE3000_DEVICEID, &reg);
 		pmic_reg_read(pfuze, PFUZE3000_REVID, &rev_id);
 		printf("PMIC: PFUZE3000 DEV_ID=0x%x REV_ID=0x%x\n",
-		       reg, rev_id);
-
+                      reg, rev_id);
 		/* disable Low Power Mode during standby mode */
 		pmic_reg_read(pfuze, PFUZE3000_LDOGCTL, &reg);
 		reg |= 0x1;
@@ -507,7 +507,6 @@ int board_mmc_init(bd_t *bis)
 	return fsl_esdhc_initialize(bis, &usdhc_cfg[1]);
 #else
 	int i, ret;
-
 	/*
 	 * According to the board_mmc_init() the following map is done:
 	 * (U-Boot device node)    (Physical Port)
@@ -651,6 +650,8 @@ static iomux_v3_cfg_t const fec1_pads[] = {
 };
 
 static iomux_v3_cfg_t const fec2_pads[] = {
+	MX6_PAD_SNVS_TAMPER6__GPIO5_IO06 | MUX_PAD_CTRL(LCD_PAD_CTRL),
+
 	MX6_PAD_GPIO1_IO06__ENET2_MDIO | MUX_PAD_CTRL(MDIO_PAD_CTRL),
 	MX6_PAD_GPIO1_IO07__ENET2_MDC | MUX_PAD_CTRL(ENET_PAD_CTRL),
 
@@ -673,12 +674,18 @@ static void setup_iomux_fec(int fec_id)
 	else
 		imx_iomux_v3_setup_multiple_pads(fec2_pads,
 						 ARRAY_SIZE(fec2_pads));
+
+	gpio_direction_output(IMX_GPIO_NR(5, 6) , 0);
+	udelay(1000);
+	gpio_direction_output(IMX_GPIO_NR(5, 6) , 1);
+	udelay(1000);
+
 }
 
 int board_eth_init(bd_t *bis)
 {
 	setup_iomux_fec(CONFIG_FEC_ENET_DEV);
-
+	
 	return fecmxc_initialize_multi(bis, CONFIG_FEC_ENET_DEV,
 				       CONFIG_FEC_MXC_PHYADDR, IMX_FEC_BASE);
 }
@@ -726,6 +733,7 @@ int board_phy_config(struct phy_device *phydev)
 
 #ifdef CONFIG_VIDEO_MXS
 static iomux_v3_cfg_t const lcd_pads[] = {
+#if 0
 	MX6_PAD_LCD_CLK__LCDIF_CLK | MUX_PAD_CTRL(LCD_PAD_CTRL),
 	MX6_PAD_LCD_ENABLE__LCDIF_ENABLE | MUX_PAD_CTRL(LCD_PAD_CTRL),
 	MX6_PAD_LCD_HSYNC__LCDIF_HSYNC | MUX_PAD_CTRL(LCD_PAD_CTRL),
@@ -760,13 +768,15 @@ static iomux_v3_cfg_t const lcd_pads[] = {
 
 	/* Use GPIO for Brightness adjustment, duty cycle = period. */
 	MX6_PAD_GPIO1_IO08__GPIO1_IO08 | MUX_PAD_CTRL(NO_PAD_CTRL),
+#endif
 };
 
 void do_enable_parallel_lcd(struct display_info_t const *dev)
 {
+#if 0
 	enable_lcdif_clock(dev->bus);
 
-	imx_iomux_v3_setup_multiple_pads(lcd_pads, ARRAY_SIZE(lcd_pads));
+	//imx_iomux_v3_setup_multiple_pads(lcd_pads, ARRAY_SIZE(lcd_pads));
 
 	/* Reset the LCD */
 	gpio_direction_output(IMX_GPIO_NR(5, 9) , 0);
@@ -775,6 +785,7 @@ void do_enable_parallel_lcd(struct display_info_t const *dev)
 
 	/* Set Brightness to high */
 	gpio_direction_output(IMX_GPIO_NR(1, 8) , 1);
+#endif
 }
 
 struct display_info_t const displays[] = {{
@@ -813,8 +824,7 @@ int board_init(void)
 	gd->bd->bi_boot_params = PHYS_SDRAM + 0x100;
 
 	imx_iomux_v3_setup_multiple_pads(iox_pads, ARRAY_SIZE(iox_pads));
-
-	iox74lv_init();
+	//iox74lv_init();
 
 #ifdef CONFIG_SYS_I2C_MXC
 	setup_i2c(0, CONFIG_SYS_I2C_SPEED, 0x7f, &i2c_pad_info1);
@@ -829,11 +839,11 @@ int board_init(void)
 #endif
 
 #ifdef CONFIG_FSL_QSPI
-	board_qspi_init();
+	//board_qspi_init();
 #endif
 
 #ifdef CONFIG_NAND_MXS
-	setup_gpmi_nand();
+	//setup_gpmi_nand();
 #endif
 
 	return 0;
